lms/djangoapps/grades/tests/test_course_grade_factory.py

```diff
@@ -97,56 +97,28 @@ def _assert_section_order(course_grade):
         with self.assertNumQueries(2), mock_get_score(1, 2):
             _assert_read(expected_pass=False, expected_percent=0)  # start off with grade of 0
 
-        # TODO: Remove Django 1.11 upgrade shim
-        # SHIM: Django 1.11 results in a few more SAVEPOINTs due to:
-        # https://github.com/django/django/commit/d44afd88#diff-5b0dda5eb9a242c15879dc9cd2121379L485
-        if django.VERSION >= (1, 11):
-            num_queries = 37
-        else:
-            num_queries = 29
-
+        num_queries = 40
```

Comment:
Can you explain why the number of queries went up by 3 here and also below?
Additionally, is it a constant number increase?  Are you reassured that it's not reflecting a linear increase in queries?

Reply:
Once for the SELECT in the new get_or_create call, once for the `recently_created` query and once for a SAVEPOINT (I think).